cluster-autoscaler/cluster_autoscaler.go

```diff
@@ -303,9 +304,21 @@ func run(_ <-chan struct{}) {
 					}
 				}
 
+				// Protect nodes from being terminated unexpectedly. Unexpected termination usually involves decreasing the node group size
```

Comment:
You can enable instance protection on the whole group but it would only [apply for new instances](https://github.com/aws/aws-sdk-go/blob/master/service/autoscaling/api.go#L10959-L10961) which doesn't cover all cases. It would also mean that new instances would have instance protection even when you disable cluster-autoscaler which I'm assuming we don't want.